proposals/templates/proposals/study_consent.html

```diff
@@ -5,138 +5,150 @@
 {% load uil_filters %}
 
 {% block header_title %}
-    {% trans "Informed consent formulieren" %} - {{ block.super }}
+  {% trans "Informed consent formulieren" %} - {{ block.super }}
 {% endblock %}
 
 {% block html_head %}
-    <script>
-        $(function() {
-            {# We use a counter so that in case we need more than 2 extra forms, this piece of code just works without changes #}
-            {% counter extra_form_counter create 1 %}
-
-            {% for form in formset %}
-                add_title('{{ form.prefix }}-informed_consent', 'Informed consent participant')
-                add_title('{{ form.prefix }}-director_consent_declaration', '{% trans 'Toestemmingsverklaring van de schoolleider/hoofd van het departement' %}')
-                {% if not form.instance.study.needs_additional_external_forms %}
-                    depends_on_hidden_value('{{ form.prefix }}-passive_consent', 'True', '{{ form.prefix }}-director_consent_declaration', true);
-                    depends_on_hidden_value('{{ form.prefix }}-passive_consent', 'True', '{{ form.prefix }}-director_consent_information');
-                {% endif %}
-                depends_on_hidden_value('{{ form.prefix }}-passive_consent', 'True', '{{ form.prefix }}-parents_information');
-                depends_on_hidden_value('{{ form.prefix }}-passive_consent', ['False', '', 'None'], '{{ form.prefix }}-informed_consent', true);
-                depends_on_hidden_value('{{ form.prefix }}-passive_consent', ['False', '', 'None'], '{{ form.prefix }}-briefing');
-
-                {% if not form.instance.study %}
-                    if($("#extra-{% counter extra_form_counter value %}").attr('data-used') == 0)
-                        $("#extra-{% counter extra_form_counter value %}").hide()
-
-                    {% counter extra_form_counter increment 1 %}
-                {% endif %}
-            {% endfor %}
-
-            // This code handles showing a new set of forms by clicking the add link
-            $("a.new").click(function ( event ) {
-                // Prevent the link from firing
-                event.preventDefault();
-
-                // Get the form number
-                var self = $(this);
-                var n = parseInt(self.attr('data-n'));
-                // Hide the clicked button
-                self.hide();
-
-                // Show the new form, and a new add link (if present, can be run without effect)
-                $("#extra-" + n).show();
-                $("#new-" + (n + 1)).show();
-
-                return false;
-            });
-
-            // Show the first add link
-            $("a.new").first().show();
-        });
-    </script>
+  <script>
+    // This function hides school-related extra forms
+    function hide_school_forms(prefix) {
```

Comment:
So, weirdly this does not work at all on my machine... I have tried this with Intervention, Observation, and Session studies, selected a school setting, nothing I did made this section appear on the page. I have tried some rudimentary debugging and the problem as far as I could trace it lead me to the `research_settings_contains_schools(self):` method in studies/models.py (line 386). It seems to be that there are no `settings_contains_schools` attributes for intervention or observation objects, nor is there a `setting__is_school` attribute in the session_set objects (also that attribute name contains a double underscore I don't know if that's a typo). It's 17:30 here and I am fried so that is how far I'm going to take this for today ;-)

Reply:
Double underscores in ORM filters can be seen as a ``.``, thus it would be equivalent to ``setting.is_school`` in python. 
Basically, you cannot use periods in parameter names (or names in general) but it is needed for nested structures, so Django implemented this instead. 
It's a bit more involved than that, but this explanation suffices for 99% of cases. 

The ``setting`` attribute does exists on the ``SettingModel`` model (and thus by extension, the intervention/observation models), and links to a model that has the ``is_school`` attribute. 


As for why it does not work? That's just plain weird honestly; the actual logic for that decision barely changed in this PR, and in fact were made less strict. So that does require some more debugging I think?

Just as an aside, looking at this template it might be a good idea to refactor (most of) the show/hide logic such that it happens inside the Django Form instead of the template. Just an idea...

Reply:
@Meesch Could you double check in your local Django admin that some of your Settings have the `is_school` attribute set? It might be called something like "Needs external permission" as a verbose name.

What we did to prod to hide all these forms is uncheck `is_school` from all settings. That "solution" potentially still prevents the forms from ever showing up.

Reply:
yeeeees that's it! "Needs external permission" has been turned off for all the settings, including school. When I turned it on, it works! I see that is also how it is saved on prod. Should we include that in the fixture then?